### PR TITLE
Configurable SSH cipher suite

### PR DESCRIPTION
**Pull request by spacetourist**
Update provides user with option to configure the cipher suite used for the built in SSH server

### CODE REVIEW

File: modules/setting/setting.go [hunk 1]
```diff
@@ -424,6 +425,7 @@ func NewContext() {
 		LandingPageURL = LANDING_PAGE_HOME
 	}
 
+	SSH.ServerCiphers = sec.Key("SSH_SERVER_CIPHERS").Strings(",")
```
Comment (Unknwon) [inline modules/setting/setting.go]: To keep in order, please move this line of code below the line `SSH.RootPath = path.Join(homeDir, ".ssh")`.

File: conf/app.ini [hunk 2]
```diff
@@ -116,6 +116,8 @@ SSH_LISTEN_HOST = 0.0.0.0
 SSH_LISTEN_PORT = %(SSH_PORT)s
 ; Root path of SSH directory, default is '~/.ssh', but you have to use '/home/git/.ssh'.
 SSH_ROOT_PATH =
+; Choose the ciphers to support for SSH connections
+SSH_SERVER_CIPHERS = aes128-ctr, aes192-ctr, aes256-ctr, aes128-gcm@openssh.com, arcfour256, arcfour128
```
Comment (Unknwon) [inline conf/app.ini]: Have you found any references that declares what ciphers that Go's ssh package supports?

Comment (spacetourist) [inline conf/app.ini]: Hi Unknwon - nice to meet you!

Issue #4107 contains an excerpt from the go source for the ssh package, this is the reference i have been working with, i presumed it to be comprehensive.

Let me know if theres anything else and please see revised commit below

**Comment by Unknwon** [comment]
Thanks!

**Comment by lunny** [comment]
@spacetourist could you also send a PR to github.com/go-gitea/gitea ?

**Comment by spacetourist** [comment]
Hi lunny, yeah, happy to do that but i'm new here. Does that mean cloning your repo, making the changes and repeating the process?

**Comment by lunny** [comment]
I think you can add a new remote gitea ref to github.com/go-gitea/gitea and create a new feature branch on your fork and cherry-pick to this new feature branch.

**Comment by spacetourist** [comment]
Phew, i think i've done it - not as straightforward as one might have hoped - you've drifted away from the tree! 

P.S. what are the core differences between gogs and gitea? Thats not exactly clear from your readme

**Comment by lunny** [comment]
Thanks

**Comment by aaronjwood** [comment]
Can I suggest that you prioritize AES-GCM above everything else? I think an authenticated encryption mode should be first on everyone's list.

It might be nice to prevent the use of the arcfour suites entirely. Since most modern chips have the AES-NI instruction set to accelerate AES operations most people would see better performance too.

**Comment by spacetourist** [comment]
I wholly agree with Aaron, certainly that was the driver behind the change in the first place and I doubt there are many cases where the arc ciphers are preferred by anyone other than those involved in downgrade attacks... Having the facility to define the suite used on a system allows those people who might wish to use the arc ciphers to add them back in however as a new contributor I certainly didn't want to impose my priorities on other users of the platform.

**Pull request merged**